packages/shared/src/components/filters/ContentTypesFilter.tsx

```diff
@@ -1,40 +1,172 @@
+import { labels } from '../../lib';
+import { useFeature } from '../GrowthBookProvider';
+import { feature } from '../../lib/featureManagement';
+import { OtherFeedPage, RequestKey } from '../../lib/query';
+import { useLogContext } from '../../contexts/LogContext';
+import { LogEvent, TargetType } from '../../lib/log';
 
 export function ContentTypesFilter(): ReactElement {
+  const postTitleLanguageFeature = useFeature(feature.postTitleLanguage);
+  const queryClient = useQueryClient();
   const { advancedSettings, isLoading } = useFeedSettings();
   const { selectedSettings, onToggleSettings } = useAdvancedSettings();
+  const { user, updateUser } = useAuthContext();
+  const { displayToast } = useToastNotification();
+  const { logEvent } = useLogContext();
 
   const videoSetting = getVideoSetting(advancedSettings);
 
+  const { mutate: onLanguageChange } = useMutation(
+    async (value?: ContentLanguage) => {
+      await updateUser({
+        ...user,
+        language: value,
+      });
+
+      await gqlClient.request(UPDATE_USER_PROFILE_MUTATION, {
+        data: {
+          language: value,
+        },
+      });
+    },
+    {
+      onMutate: (value) => {
+        logEvent({
+          event_name: LogEvent.ChangeSettings,
+          target_type: TargetType.Language,
+          target_id: value,
+        });
+      },
+      onSuccess: async () => {
+        const requestKeys = [
+          ...Object.values(SharedFeedPage),
+          OtherFeedPage.Preview,
+          RequestKey.Squad,
+          RequestKey.FeedPreview,
+          RequestKey.FeedPreviewCustom,
+          RequestKey.PostKey,
+          RequestKey.Bookmarks,
+          RequestKey.ReadingHistory,
+          RequestKey.RelatedPosts,
+          RequestKey.SourceFeed,
+          RequestKey.SourceMostUpvoted,
+          RequestKey.SourceBestDiscussed,
+          RequestKey.TagFeed,
+          RequestKey.TagsMostUpvoted,
+          RequestKey.TagsBestDiscussed,
+        ];
+
+        await Promise.all(
+          requestKeys.map((requestKey) => {
+            const queryKey = [requestKey];
+
+            return queryClient.invalidateQueries({
+              queryKey,
+              exact: false,
+              type: 'all',
+            });
+          }),
+        );
+      },
+      onError: () => {
+        displayToast(labels.error.generic);
+      },
+    },
+  );
+
   return (
-    <section className="flex flex-col gap-6 px-6" aria-busy={isLoading}>
-      {videoSetting && (
-        <FilterCheckbox
-          name={videoSetting.title}
-          description={videoSetting.description}
-          checked={
-            selectedSettings[videoSetting.id] ??
-            videoSetting.defaultEnabledState
-          }
-          onToggle={() =>
-            onToggleSettings(videoSetting.id, videoSetting.defaultEnabledState)
-          }
-        >
-          {videoSetting.title}
-        </FilterCheckbox>
+    <div className="flex flex-col gap-4 px-6">
```

Comment:
Overall space here feels too big, should only be 24px allround.
![Screenshot 2024-08-21 at 15 52 17](https://github.com/user-attachments/assets/e059f41b-2797-4995-b2b5-a361eece9ad1)

Reply:
I left it as is on other filter views. Top one is actually a bit smaller the other. I can ping design but if you agree I can do engineering decision and just put 24 as you say on all of them?

Reply:
Ah just put 24 on all for now then :)

Reply:
https://github.com/dailydotdev/apps/pull/3474